File: settlement/room/infra/importt/ROOM_IMPORT.java
package settlement.room.infra.importt;

import java.io.IOException;

import init.resources.RESOURCE;
import settlement.path.finders.SFinderRoomService;
import settlement.room.main.RoomBlueprintIns;
import settlement.room.main.category.RoomCategorySub;
import settlement.room.main.furnisher.Furnisher;
import settlement.room.main.util.RoomInitData;
import snake2d.util.file.FileGetter;
import snake2d.util.file.FilePutter;
import snake2d.util.sets.LISTE;
import view.sett.ui.room.UIRoomModule;

public final class ROOM_IMPORT extends RoomBlueprintIns<ImportInstance>{

	public final ImportTally tally = new ImportTally();
	final Crate crate = new Crate(this);
	
	final Constructor constructor;
	
	public ROOM_IMPORT(RoomInitData init, RoomCategorySub cat) throws IOException {
		super(0, init, "_IMPORT", cat);
		
		constructor = new Constructor(this, init);
	}

	@Override
	protected void update(double ds) {

	}
	
	@Override
	public Furnisher constructor() {
		return constructor;
	}

	@Override
	public SFinderRoomService service(int tx, int ty) {
		return null;
	}
	
	@Override
	protected void saveP(FilePutter saveFile){
		tally.saver.save(saveFile);
	}
	
	@Override
	protected void loadP(FileGetter saveFile) throws IOException{
		tally.saver.load(saveFile);
		for (int i = 0; i < instancesSize(); i++) {
			tally.count(getInstance(i).resource(), getInstance(i).amount(), getInstance(i).capacity());
		}
	}
	
	@Override
	protected void clearP() {
		tally.saver.clear();
	}
	
	public final ImportThingy UNLOADER = new ImportThingy(this, tally);
	
	@Override
	public void appendView(LISTE<UIRoomModule> mm) {
		mm.add(new Gui(this).make());
	}
	
	public int getBestPrice(RESOURCE res) {
		return 0;
	}
	
	
}

File: LegacySource/settlement/room/infra/importt/ROOM_IMPORT.cs
using System;
using System.Collections.Generic;
using System.IO;
using Init.Resources;
using Settlement.Path.Finders;
using Settlement.Room.Main;
using Settlement.Room.Main.Category;
using Settlement.Room.Main.Furnisher;
using Settlement.Room.Main.Util;
using Snake2D.Util.File;
using Snake2D.Util.Sets;
using View.Sett.UI.Room;

namespace Settlement.Room.Infra.Importt
{
    public sealed class ROOM_IMPORT : RoomBlueprintIns<ImportInstance>
    {
        public readonly ImportTally tally = new ImportTally();
        private readonly Crate crate = new Crate(this);

        private readonly Constructor constructor;

        public ROOM_IMPORT(RoomInitData init, RoomCategorySub cat) : base(0, init, "_IMPORT", cat)
        {
            constructor = new Constructor(this, init);
        }

        protected override void Update(double ds)
        {
        }

        public override Furnisher Constructor()
        {
            return constructor;
        }

        public override SFinderRoomService Service(int tx, int ty)
        {
            return null;
        }

        protected override void SaveP(FilePutter saveFile)
        {
            tally.saver.Save(saveFile);
        }

        protected override void LoadP(FileGetter saveFile)
        {
            tally.saver.Load(saveFile);
            for (int i = 0; i < InstancesSize(); i++)
            {
                tally.Count(GetInstance(i).Resource(), GetInstance(i).Amount(), GetInstance(i).Capacity());
            }
        }

        protected override void ClearP()
        {
            tally.saver.Clear();
        }

        public readonly ImportThingy UNLOADER = new ImportThingy(this, tally);

        public override void AppendView(LISTE<UIRoomModule> mm)
        {
            mm.Add(new Gui(this).Make());
        }

        public int GetBestPrice(RESOURCE res)
        {
            return 0;
        }
    }
}